File: src/main/java/com/mikea/gae/rx/RxHttpRequestEvent.java
package com.mikea.gae.rx;

import javax.servlet.http.HttpServletRequest;
import java.io.IOException;

/**
 * @author mike.aizatsky@gmail.com
 */
public class RxHttpRequestEvent {
    private final Rx rx;
    public final HttpServletRequest request;
    private final RxHttpResponse response;

    protected RxHttpRequestEvent(RxHttpRequestEvent event) {
        this(event.rx, event.request, event.response);
    }

    public RxHttpRequestEvent(Rx rx, HttpServletRequest request, RxHttpResponse response) {
        this.rx = rx;
        this.request = request;
        this.response = response;
    }

    public void sendRedirect(String url) throws IOException {
        response.sendRedirect(url);
    }


    public void sendError(int i, String message) throws IOException {
        response.sendError(i, message);
    }

    public void sendOk() throws IOException {
        response.sendOk();
    }
}

File: src/main/scala/com/mikea/gae/rx/RxHttpRequestEvent.scala
package com.mikea.gae.rx

import javax.servlet.http.HttpServletRequest

/**
 * @author mike.aizatsky@gmail.com
 */
class RxHttpRequestEvent(rx: Rx, _request: HttpServletRequest, _response: RxHttpResponse) {
  def sendRedirect(url: String): Unit = {
    response.sendRedirect(url)
  }

  def sendError(i: Int, message: String): Unit = {
    response.sendError(i, message)
  }

  def sendOk(): Unit = {
    response.sendOk()
  }

  def request = _request
  def response = _response
}